Retina display fixes

schm1dtmac opened the pull request:
Changes are based on cmaughan's fixes from https://github.com/ocornut/imgui/issues/3757#issuecomment-800921198 but messed with a bit to work on Xemu's ImGUI fork. The result is that ImGUI won't render at a low-DPI resolution anymore and now should provide crisp UI on Retina Macs (although the in-built UI scaling option in Xemu will need to be set to 2x to ensure the UI isn't tiny).

mborgerson committed:
Disable NavUpdateWindowing

schm1dtmac committed:
Retina display fixes

Changes are based on cmaughan's fixes from https://github.com/ocornut/imgui/issues/3757#issuecomment-800921198 but messed with a bit to work on Xemu's ImGUI fork

Commit:
Merge branch 'xemu-project:xemu' into xemu

schm1dtmac commented:
@mborgerson Sorry to prod you about this after so long, is there anything holding this back from being merged?

mborgerson commented:
I'll take a look shortly!

mborgerson commented on backends/imgui_impl_sdl2.cpp:
```diff
@@ -620,7 +620,27 @@ static void ImGui_ImplSDL2_UpdateMouseData()
+            
+            if (display_h != h)
+            {
+                io.AddMousePosEvent((float)2*(mouse_x_global - window_x), (float)2*(mouse_y_global - window_y));
```
Assumes scale is always 2x?

schm1dtmac replied:
macOS only supports 1x and 2x scale factors and this looks unlikely to change (other scales are achieved via GPU upscaling/downscaling), so 2x is pretty safe bet in the case where we're not at 1x scale. ~~Attempting to get the actual backing scale factor rigorously requires calling some ObjC methods which wouldn't be as nice but possible if you'd prefer.~~

schm1dtmac replied:
Latest commit should address this properly using the ratio between SDL window and drawable size @mborgerson.
EDIT: Verified this all continues to work fine locally too, no issues so provided this was the only concern, LGTM to me.

Commit:
Address review comments

Whilst the scaling factor should only ever be 1x/2x on macOS this change makes it a bit more rigorous now.

mborgerson commented:
Please create necessary PR for xemu and confirm that it works as expected. Thanks!

schm1dtmac commented:
> Please create necessary PR for xemu and confirm that it works as expected. Thanks!

@mborgerson No problem, filed the needed imgui update PR at https://github.com/xemu-project/xemu/pull/2637, if you can approve the workflow/CI that'd be great thanks.
